### PR TITLE
[POC][DO NOT MERGE] example of delta table/autoloader support for databricks users

### PR DESCRIPTION
**Pull request by dimberman**
…elta-table-support# Description
## What is the new behavior?

-
-
-

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by feluelle — COMMENTED** [review]
I looked up what the [cli](https://github.com/databricks/databricks-cli#readme) does.

> The Databricks Command Line Interface (CLI) is an open source tool which provides an easy to use interface to the Databricks platform. The CLI is built on top of the Databricks REST APIs.
> 
> Note: This CLI is under active development and is released as an experimental client. This means that interfaces are still subject to change.

It also uses the REST API under the hood (which is expected), which we already use inside `DatabricksHook`. In my opinion we should either use the provider or the cli - but not both.

I am personally in favor of using the provider, because that should be our interface to external libraries. Especially because the CLI is still in experimental stage. So changes can easily break the code if we are not pinning the version or defining the version requirements carefully. WDYT? @dimberman 

Once the cli is stable, the ideal solution would then be to make the changes in airflow providers to use the cli instead of the REST api.

File: python-sdk/pyproject.toml [hunk 1]
```diff
@@ -75,11 +75,18 @@ amazon = [
 ]
 openlineage = ["openlineage-airflow>=0.15.1"]
 
+databricks = [
+    "apache-airflow-providers-databricks",
+    "databricks-cli"
```
[the remark this answers is not in the document]
Comment (dimberman) [reply python-sdk/pyproject.toml:80]: @feluelle yes the `databricks-cli` library is where the SDK lives. Don't ask me why.

**Comment by dimberman** [comment]
@feluelle the provider doesn't have all of the endpoints that are used her (e.g. DBFS, secrets, etc) so I would consider it as insufficient for our use-case. The main reason I have downloaded the provider at all is for access to the DatabricksSqlHook. Glad to discuss further but I'd rather use an officially supported python SDK where possible.

File: python-sdk/pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (feluelle) [reply python-sdk/pyproject.toml:80]: Don't ask me why we have `sql-cli` in `astro-sdk` 🤣

File: python-sdk/src/astro/spark/autoloader/autoloader_job.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import tempfile
+import time
+from pathlib import Path
+from typing import Dict
+
+from databricks_cli.dbfs.api import DbfsApi, DbfsPath
+from databricks_cli.jobs.api import JobsApi
+from databricks_cli.runs.api import RunsApi
+from databricks_cli.sdk.api_client import ApiClient
+from databricks_cli.secrets.api import SecretApi
```
[the remark this answers is not in the document]
Comment (feluelle) [reply python-sdk/src/astro/spark/autoloader/autoloader_job.py:12]: Yes, but we should then change this to use the cli too, right?

**Comment by feluelle** [comment]
I am fine with that as a first step. We can iterate and improve later. I just wanted to bring this up. :)

**Comment by tatiana** [comment]
@dimberman would it make sense to close the PoC PR, since you're making production-ready ones?

**Pull request closed without merging**